internal/pkg/util/fs/helper.go

```diff
@@ -202,6 +212,7 @@ func CopyFile(from, to string, mode os.FileMode) (err error) {
 
 	_, err = io.Copy(dstFile, srcFile)
 	if err != nil {
+		os.Remove(to)
```

Comment:
I suggest you to keep defer func and use named return parameters as discussed. 
You need to close dstFile prior to removing it.